src/main/java/com/continuum/nova/gui/NovaDraw.java

```diff
@@ -62,11 +76,12 @@ public static void draw(ResourceLocation texture, Integer[] indexBuffer, Float[]
      * @param vertices    the vertices as Vertex objects
      */
     public static void draw(ResourceLocation texture, Integer[] indexBuffer, Vertex[] vertices) {
+
         Float[] vertexbuffer = new Float[vertices.length * 9];
         for (int v = 0; v < vertices.length; v++) {
             vertexbuffer[v * 9] = vertices[v].x;
             vertexbuffer[v * 9 + 1] = vertices[v].y;
-            vertexbuffer[v * 9 + 2] = vertices[v].z;
+            vertexbuffer[v * 9 + 2] = currentZ;
```

Comment:
Since we're not getting the z from `Vertex` objects, let's cascade this change throughout NovaDraw